docs/features/wait/host_port.md

```diff
@@ -0,0 +1,31 @@
+# HostPort wait strategy
+
+The host-port wait strategy will check if a port is listening in the container, being able to set the following conditions:
+
+- a port to be listening in the container. The port and protocol to be used, which is represented by a string containing port number and protocol in the format "80/tcp".
+- the first exposed port in the container.
+- the startup timeout to be used, default is 60 seconds.
+- the poll interval to be used, default is 100 milliseconds.
+
+Variations on the HosPort wait strategy are supported, including:
+
+## Listening port in the container
+
+```golang
+req := ContainerRequest{
+    Image:        "docker.io/nginx:alpine",
+    ExposedPorts: []string{"80/tcp"},
+    WaitingFor:   wait.ForListeningPort("80/tcp"),
+}
+```
+
+## First exposed port in the container
+
+The wait strategy will use the first exposed port from the image configuration.
```

Comment:
The first exposed port at the Dockerfile: the code gets the image config directly from Docker, and retrieves the first port from there.

Reply:
So if I expose a port in the `ContainerRequest`, this would not be considered?

Reply:
@kiview I've double checked the code and when the `wait.ForExposedPort` function is used, then it will use the first port from raw representation of the container:

container.NetworkSettings.Ports -> array of nat.Port

If the array is empty, then an error is thrown, which makes me consider a possible unexpected behavior in the case somebody used this method AND there is an exposed port.

Because it's clear in the code (https://github.com/testcontainers/testcontainers-go/blob/6ee7f6a952f3e377de0d70b3b8bc81b3abe5a53d/wait/host_port.go#L68-L86), I think I'm going to explain more in depth this behavior in the docs

Reply:
Oh, yeah I realized it's "container" definition, not "image" definition. So if the user exposed a port in the container request struct, but used the `ForExposedPort` func, then the container definition will include it, therefore the `Ports` call will retrieve it. Thanks for pointing that out.

Nevertheless, let me explain the behavior in the docs